### PR TITLE
Removes Stu3 from Sql provider

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
Makes the Sql provider version agnostic

## Related issues
Addresses [AB#69127](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/69127).

## Testing
All existing unit, integration and e2e tests must pass

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs [hunk 1]
```diff
@@ -15,12 +15,16 @@ public class SearchParameterInfo
             string name,
             Uri url,
             string searchParamType,
-            ICollection<SearchParameterComponentInfo> components)
+            IReadOnlyList<SearchParameterComponentInfo> components,
+            string expression,
+            IReadOnlyCollection<string> targetResourceTypes)
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs]: null check?

File: src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs [hunk 2]
```diff
@@ -106,7 +107,7 @@ private Expression ParseChainedExpression(string resourceType, SearchParameter s
 
             // We have more paths after this so this is a chained expression.
             // Since this is chained expression, the expression must be a reference type.
-            if (searchParameter.Type != SearchParamType.Reference)
+            if (searchParameter.Type != SearchParamType.Reference.ToString())
```
Comment (johnstairs) [inline src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs]: Can we make an enum this?

Comment (johnstairs) [reply src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs]: Looks like a lot of `ToString()`ing

File: src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs (hunk 1 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Core/Models/SearchParameterInfo.cs]: A lot of these seem to be optional, how do you feel about giving them defaults (i.e. expression = null) or allow property setters?

**Pull request merged**